### PR TITLE
Add support for Authentication against Azure Key Vault using Client Certificate

### LINKED ISSUE #3461: Support for Azure KV Authentication with ClientCertificate
**Is your feature request related to a problem? Please describe.**
I find that the operator should be able to authenticate via ClientCredentials. This is an easy implementation since the used kvauth package already supports this: https://pkg.go.dev/github.com/Azure/go-autorest/autorest/azure/auth#NewClientCertificateConfig

**Describe the solution you'd like**
In the perfect world, it should be possible to configure a `SecretStore` and a `ClusterSecretStore` to fetch the clientCertificate key from a secret and use that for authentication against Azure KV rather than the clientSecret field.

**Describe alternatives you've considered**
None

**Additional context**
This is how I want the CRD to work:

```yaml
apiVersion: external-secrets.io/v1beta1
kind: ClusterSecretStore
metadata:
  name: store
spec:
  provider:
    azurekv:
      authSecretRef:
        clientId:
          key: ClientID
          name: azure-secret-sp
        clientCertificate:
          key: ClientCertificate
          name: azure-secret-sp
      authType: ServicePrincipal
      environmentType: PublicCloud
      tenantId: <omitted>
      vaultUrl: "https://<kv>.vault.azure.net/"
```

If somebody is willing to point me to the relevant code locations, I am very happy to contribute this myself, if others agree that this would be a good feature!

### PR DESCRIPTION
**Pull request by lieberlois**
## Problem Statement

I find that the operator should be able to authenticate via ClientCredentials. In the perfect world, it should be possible to configure a SecretStore and a ClusterSecretStore to fetch the clientCertificate key from a secret and use that for authentication against Azure KV rather than the clientSecret field.

## Related Issue

Fixes #3461 

## Proposed Changes

Extend the CRD to not only accept clientSecret but also clientCertificate

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test` -- main branch is currently failing, the tests that i changed are all green ✅ 
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/provider/azure/keyvault/keyvault.go [hunk 1]
```diff
@@ -220,6 +222,12 @@ func (a *Azure) ValidateStore(store esv1beta1.GenericStore) (admission.Warnings,
 				return nil, fmt.Errorf(errInvalidSecRefClientSecret, err)
 			}
 		}
+		// if p.AuthSecretRef.ClientCertificate != nil {
+		// 	path := p.AuthSecretRef.ClientCertificate.Path
+		// 	if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+		// 		return nil, fmt.Errorf(errInvalidSecRefClientCertificate, err)
+		// 	}
+		// }
```
Comment (gusfcarvalho) [inline pkg/provider/azure/keyvault/keyvault.go:230]: I think this needs to be removed

File: pkg/provider/azure/keyvault/keyvault.go [hunk 2]
```diff
@@ -990,20 +1001,60 @@ func (a *Azure) authorizerForServicePrincipal(ctx context.Context) (autorest.Aut
...
+		if err != nil {
+			return nil, err
+		}
+
+		return getAuthorizerForClientCertificate(
+			clientID,
+			[]byte(clientCertificate),
+			*a.provider.TenantID,
+			a.provider.EnvironmentType,
+		)
```
Comment (gusfcarvalho) [inline pkg/provider/azure/keyvault/keyvault.go:1040]: can we move this block into their own functions to reduce complexity a bit?

**Comment by gusfcarvalho** [comment]
hi @lieberlois thanks for your contribution! Just some nits in the PR - plus, if you can run `make reviewable` - some linting tests are failing 😄

File: pkg/provider/azure/keyvault/keyvault.go (hunk 1 above)
Comment (lieberlois) [reply pkg/provider/azure/keyvault/keyvault.go:230]: Yupp, sorry, forgot that 😉

File: pkg/provider/azure/keyvault/keyvault.go (hunk 2 above)
Comment (lieberlois) [reply pkg/provider/azure/keyvault/keyvault.go:1040]: Sure, got it 👍

**Comment by lieberlois** [comment]
@gusfcarvalho Thanks for the review! Maybe we need to extend the developer docs - for MacOS (at least arm64 chips) you need to have gnu-utils installed and cannot use the MacOS native sed etc. 

I applied the reviewed changes 👍

**Comment by gusfcarvalho** [comment]
> @gusfcarvalho Thanks for the review! Maybe we need to extend the developer docs - for MacOS (at least arm64 chips) you need to have gnu-utils installed and cannot use the MacOS native sed etc. - this is why i couldnt run `make reviewable` before ^^
> 
> I applied the reviewed changes 👍

We used to have as part of our make files the detection of macOS and bump of the appropriate `sed` commands to use. Maybe this needs to be reviewed once again 🤔

**Comment by gusfcarvalho** [comment]
hi @lieberlois ! I think we only need now to fix the message from that unit test: https://github.com/external-secrets/external-secrets/actions/runs/9042682072/job/24850275469?pr=3469#step:7:940

**Comment by lieberlois** [comment]
Hey @gusfcarvalho 😉  yeah, that was a search and replace error, I corrected it. Tests are now running 😄  Sorry for missing that !

**Comment by lieberlois** [comment]
> > @gusfcarvalho Thanks for the review! Maybe we need to extend the developer docs - for MacOS (at least arm64 chips) you need to have gnu-utils installed and cannot use the MacOS native sed etc. - this is why i couldnt run `make reviewable` before ^^
> > I applied the reviewed changes 👍
> 
> We used to have as part of our make files the detection of arm64 platforms and bump of the appropriate `sed` commands, install of needed tools, etc, to use. Maybe this needs to be reviewed once again 🤔


Regarding this topic: yes, the detection works, i just didnt have "gnu-sed" on my mac (which basically aliases "gsed" to the gnu-sed version). The script however requires this.

**Review by gusfcarvalho — APPROVED** [review]

**Pull request merged**